packages/ra-ui-materialui/src/list/types.ts

```diff
@@ -1,6 +1,6 @@
 import { Identifier, RaRecord } from 'ra-core';
 
-export type RowClickFunction = <RecordType extends RaRecord = RaRecord>(
+export type RowClickFunction<RecordType extends RaRecord = RaRecord> = (
```

Comment:
We use this in `SimpleListItem` from `SimpleList` too. Can you make it accept a `RecordType` as well?